base_business_document_import/models/business_document_import.py

```diff
@@ -0,0 +1,1398 @@
+            "fixed",
+            "percent",
+        ], "bad tax type, has to be fixed or percent"
+        assert "amount" in tax_dict, "Missing amount key in tax_dict"
+        domain.append(("amount_type", "=", tax_dict["amount_type"]))
+        if tax_dict.get("unece_type_code"):
+            domain.append(("unece_type_code", "=", tax_dict["unece_type_code"]))
+        if tax_dict.get("unece_categ_code"):
+            domain.append(("unece_categ_code", "=", tax_dict["unece_categ_code"]))
+        if tax_dict.get("unece_due_date_code"):
+            tax_exigibility = ato._get_tax_exigibility_from_unece_code(
+                tax_dict["unece_due_date_code"]
+            )
+            if tax_exigibility:
+                domain = expression.AND(
+                    [domain, [("tax_exigibility", "=", tax_exigibility)]]
+                )
+        return domain
+
+    @api.model
+    def _match_tax(
+        self,
+        tax_dict,
+        chatter_msg,
+        type_tax_use="purchase",
+        price_include=False,
+    ):
+        """Example:
+        tax_dict = {
+            'amount_type': 'percent',  # required param, 'fixed' or 'percent'
+            'amount': 20.0,  # required
+            'unece_type_code': 'VAT',
+            'unece_categ_code': 'S',
+            'unece_due_date_code': '72',
+            }
+        """
+        ato = self.env["account.tax"]
+        self._strip_cleanup_dict(tax_dict)
+        tax = self._direct_match(tax_dict, ato)
+        if tax:
+            return tax
+        domain = self._prepare_match_tax_domain(
+            tax_dict, type_tax_use=type_tax_use, price_include=price_include
+        )
+        taxes = ato.search(domain)
+        for tax in taxes:
+            tax_amount = tax.amount  # 'amount' field : digits=(16, 4)
+            if not float_compare(tax_dict["amount"], tax_amount, precision_digits=4):
+                return tax
+        raise self.user_error_wrap(
+            "_match_tax",
+            tax_dict,
+            _(
+                "Odoo couldn't find any tax with 'Tax Application' = '%(tax)s' "
+                "and 'Tax Included in Price' = '%(price)s' which correspond to the "
+                "following information extracted from the business document:\n"
+                "UNECE Tax Type code: %(tax_type)s\n"
+                "UNECE Tax Category code: %(tax_cat)s\n"
+                "UNECE Due Date code: %(tax_due_date)s\n"
+                "Tax amount: %(amount)s %(amount_type)s",
+                tax=type_tax_use,
+                price=price_include,
+                tax_type=tax_dict.get("unece_type_code") or "",
+                tax_cat=tax_dict.get("unece_categ_code") or "",
+                tax_due_date=tax_dict.get("unece_due_date_code") or "",
+                amount=tax_dict["amount"],
+                amount_type=tax_dict["amount_type"] == "percent"
+                and "%"
+                or _("(fixed)"),
+            ),
+        )
+
+    def compare_lines(
```

Comment:
Do we need add test case for this code?

Reply:
This method is added as a hook which lets other modules inherit and call. At this moment, our priority is on reviewing what we changed in this commit: https://github.com/nguyenminhchien/edi/pull/3/commits/f1b9e90099f89d4db1601c02de9a5535cfff9e29